apps/web/src/components/Feed/Socialize/CommentsModal/FeedEventsCommentsModal.tsx

```diff
@@ -0,0 +1,120 @@
+
+import { FeedEventsCommentsModalFragment$key } from '~/generated/FeedEventsCommentsModalFragment.graphql';
+import { FeedEventsCommentsModalQueryFragment$key } from '~/generated/FeedEventsCommentsModalQueryFragment.graphql';
+import useCommentOnFeedEvent from '~/hooks/api/feedEvents/useCommentOnFeedEvent';
+import getVideoOrImageUrlForNftPreview from '~/shared/relay/getVideoOrImageUrlForNftPreview';
+
+import { CommentsModal } from './CommentsModal';
+
+type Props = {
+  eventRef: FeedEventsCommentsModalFragment$key;
+  queryRef: FeedEventsCommentsModalQueryFragment$key;
+  fullscreen: boolean;
+};
+
+export function FeedEventsCommentsModal({ eventRef, queryRef, fullscreen }: Props) {
+  const {
+    data: feedEvent,
+    loadPrevious,
+    hasPrevious,
+  } = usePaginationFragment(
+    graphql`
+      fragment FeedEventsCommentsModalFragment on FeedEvent
+      @refetchable(queryName: "FeedEventsCommentsModalRefetchableFragment") {
+        interactions(last: $interactionsFirst, before: $interactionsAfter)
+          @connection(key: "CommentsModal_interactions") {
+          edges {
+            node {
+              __typename
+
+              ... on Comment {
+                ...CommentsModalFragment
+              }
+            }
+          }
+        }
+        id
+        dbid
+      }
+    `,
+    eventRef
+  );
+
+  const query = useFragment(
+    graphql`
+      fragment FeedEventsCommentsModalQueryFragment on Query {
+        viewer {
+          ... on Viewer {
+            user {
+              id
+              dbid
+              username
+              profileImage {
+                ... on TokenProfileImage {
+                  token {
+                    dbid
+                    id
+                    ...getVideoOrImageUrlForNftPreviewFragment
+                  }
+                }
+              }
+            }
+          }
+        }
+        ...CommentsModalQueryFragment
+      }
+    `,
+    queryRef
+  );
+
+  const nonNullInteractions = useMemo(() => {
+    const interactions = [];
+
+    for (const interaction of feedEvent.interactions?.edges ?? []) {
+      if (interaction?.node && interaction.node.__typename === 'Comment') {
+        interactions.push(interaction.node);
+      }
+    }
+
+    return interactions.reverse();
+  }, [feedEvent.interactions?.edges]);
+
+  const [commentOnFeedEvent, isSubmittingComment] = useCommentOnFeedEvent();
+
+  const handleSubmitComment = useCallback(
+    (comment: string) => {
+      const { token } = query.viewer?.user?.profileImage ?? {};
+      const user = query.viewer?.user;
+      if (!user) {
+        return;
+      }
+
+      const result = token
+        ? getVideoOrImageUrlForNftPreview({
+            tokenRef: token,
+          })
+        : null;
+      commentOnFeedEvent(feedEvent.id, feedEvent.dbid, comment, {
+        id: user.id,
+        dbid: user.dbid,
+        username: user.username ?? '',
+        profileImageUrl: result?.urls?.small ?? '',
+      });
```

Comment:
non-blocking nit:

seems like the main pieces of info we need to pass in are the feed event info + comment. the user-related stuff is for optimistic syncing, which could be pulled out by the helper function itself.

example:
```js
commentOnFeedEvent({
  id: feedEvent.id,
  dbid: feedEvent.dbid,
  comment: comment,
  queryRef: queryRef <-- function can handle pulling out relevant user info
})
```

the benefit would be that wherever `commentOnFeedEvent` is used, we don't have to duplicate logic that handles seemingly unrelated user properties (this logic is consolidated at the helper level)

a precedent for something like this is `getVideoOrImageUrlNftPreview`, where it reads inline data from the relay cache, even though the function itself isn't a hook

<img width="641" alt="image" src="https://github.com/gallery-so/gallery/assets/12162433/c90939b8-dd72-41f8-a1f6-aad4215c740a">